baseline.yaml

```diff
@@ -649,6 +649,28 @@ criteria:
     security_insights_value: # TODO
     scorecard_probe: # TODO
 
+      - id: OSPS-DO-19   
+    maturity_level: 3
+    category: Documentation
+    criteria: |
+       The project MUST perform an external security
+       review or audit to understand the most likely 
+       and impactful problems that could occur within the 
+       software, and make plans to address those
+       problems.
```

Comment:
I'd probably either use a time-based or functionality-based model.  A 4-year-old threat assessment for Kubernetes (still major version 1) could be from Kubernetes 1.17 (EOL 13 January 2021) and may be almost useless today.  For Google Chrome, major version updates seem to be every month or two.

```suggestion
       The project MUST have performed an external security
       review or audit within the last 4 years or after major 
       architectural changes to understand the most likely 
       and impactful problems that could occur within the 
       software, and make plans to address those
       problems.
```

At the same time, ever having a security assessment feels like it is a bit of an achievement for many projects.  I'm assuming that we view maturity level 3 as a significant accomplishment (similar difficulty to CNCF graduation?) that would be a minority of projects?

Reply:
Looking over the CRA, (39) 
> software update modifies the intended purpose of that product and those changes were not foreseen by the manufacturer in the initial risk assessment,

this would indicate anytime a change occurs to the project that is different from the original use and support of the project; like expanded scope, brand new functionality.  This would match up with every major release (depending on how the project is defining major releases) as well as those major changes which should trigger a new threat model. 

difference here is not just added functionality, it is the removal of functionality, or alternative functionality that achieves the same outcome but may do so in a manner that alters the flow, operation, or actions to achieve the same outcome.

Reply:
yes - external security audits is a huge achievement for projects and usually only occurs when interested parties are willing to fund it, which is usually gated within open source foundations by reaching a substantial level of maturity (aka production ready) where the results of the audit provides the most value to adopters of the project.

a security assessment may not be the same as an audit in all cases. In the context of open source projects (taking the CNCF example) joint-assessments are performed by community members in the ecosystem who may or may not be (by job title/role) a "security professional". We're very clear that these assessments do not replace an audit, and make no guarantees as to the depth of coverage.  i.e. we have a placeholder for hands on testing of the project, but only if we have a community member with the skills to do it, and even then we make no guarantees as to the depth and complexity of coverage in that hands on testing.